docs/guide/template-filters.md

```diff
@@ -0,0 +1,118 @@
+
+## Custom template filters
+If you have special template needs you can also construct custom filters for use in Nunjucks templates.
+
+```javascript
+// lib/modules/link-widgets/index.js
+module.exports = {
+  extend: '@apostrophecms/widget-type',
+  options: {
+    label: 'Link to a Page'
+  },
+  fields: {
+    add: {
+      url: {
+        type: 'url',
+        label: 'URL',
+        required: true
+      },
+      label: {
+        type: 'string',
+        label: 'Label',
+        required: true
+      }
+    },
+    group: {
+      links: {
+        label: 'Links',
+        fields: [ 'url', 'label' ]
+      }
+    }
+  },
+  init(self) {
+    self.apos.template.addFilter({
+      stripHttp: self.stripHttp
+    });
+  },
+  methods(self) {
+    return {
+      stripHttp(s) {
+        return s.replace(/^(https?:|)\/\//, '');
+      }
+    };
+  }
+};
+```
+To use this filter you would simply pipe your data to the filter.
+
+```markup
+// lib/modules/link-widgets/views/widget.html
+<section data-link-widget>
+  <h2>{{ data.widget.label }}: {{ data.widget.url | stripHttp }}</h2>
+</section>
+```
+
+## Alphabetical Apostrophe filter reference
+
+### `| build(obj...)`
+
+Given a URL and one or more objects, this filter adds query string parameters for each of the properties in the objects. If objects affect the same parameter, the last object wins. If a parameter is set to `null` or the empty string it is removed from the URL altogether. This is very useful for adding filters to the current URL, respecting other filters already present, without complicated logic.
+
+The `build` filter has additional features which you can read about in the comments of the `index.js` file of the url module.
+
+### `| clonePermanent`
+
+Given JavaScript data, this filter recursively strips out any properties whose names beginning with a `_`, except `_id`. This is used to avoid pushing large joined documents into the DOM.
+
+### `| css`
+
+Converts a string to a hyphenated CSS identifier. For instance, `fooBar` becomes `foo-bar`.
+
+### `| date(format)`
+
+Turns a JavaScript Date object into a string, as specified by `format`. For formatting options, see [the documentation of the `momentjs` npm module](https://momentjs.com/docs/#/displaying/format/).
+
+### `| json`
+
+Turns JavaScript data into a JSON string, **correctly escaped for safe use inside a** `script` **tag in the middle of an HTML document.** Note that it is **not** safe to use the Nunjucks `dump` filter in this way. This filter is not for attributes, see `jsonAttribute`.
+
+### `| jsonAttribute(options)`
+
+Given JavaScript data, this filter escapes it correctly to be the value of a `data-` attribute of an element in the page. It does not add the `"` quote characters, but it does escape any `"` characters in the JSON string.
```

Comment:
It does not add the `"` quote characters around the attribute itself,...

Reply:
Clarified